src/main/java/seedu/address/logic/commands/UpdateTaskCommand.java

```diff
@@ -0,0 +1,185 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_TASKS;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.task.Task;
+
+/**
+ * Updates a task to the taskList at the specified index with specified details.
+ */
+public class UpdateTaskCommand extends Command {
+    public static final String COMMAND_WORD = "updateTask";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Updates a task in the taskList\n"
+            + "Parameters: Index, Task\n"
+            + "Example: " + COMMAND_WORD + " 1 d/Purchase milk dl/2022-10-18 t/Food";
```

Comment:
Good format for example, should generalise all our examples like this from now on to show actual possible inputs rather than an expected format